plugins/rule_engines/irods_rule_language/src/functions.cpp

```diff
@@ -1970,6 +1970,33 @@ Res *smsi_remoteExec( Node** paramsr, int, Node* node, ruleExecInfo_t* rei, int,
 #endif
 }
 
+/**
+ * \fn writeLine(msParam_t *where,  msParam_t *inString, ruleExecInfo_t *rei)
+ *
+ * \brief  Writes a message to either the server log, stdout, stderr, or a data object.
+ *
+ * \module core
+ *
+ * \usage See clients/icommands/test/rules/
```

Comment:
is this correct?

Reply:
Is this a question about line 1980?  There are a lot of writeLine() examples in ~irods/client/icommands/test/rules.

Reply:
does that path really exist?   we moved icommands to a different repository...

Reply:
It exists on the filesystem under the irods admin service account.  I assumed that that is what they are referencing?

Reply:
Also, if that needs to change then it needs to be done for all of the microservices.

Reply:
Would it make more sense to refer to https://github.com/irods/irods_client_icommands/test/rules?

Reply:
Given those files are .r files, perhaps we should move them into this repo.

Seems weird to keep them in the icommands repo.

Thoughts?

Reply:
We could also just... write the usage example here. :p 
```c
writeLine("stdout", "This shows up on stdout");
writeLine("stderr", "This shows up on stderr");
writeLine("serverLog", "This shows up in the server log");
writeLine("/tempZone/home/alice/foo", "This is written to the data object /tempZone/home/alice/foo, I guess");
```

Reply:
I'm thinking we replace the links with concrete examples, which can happen later.

Reply:
I can add those.  I'm not sure how it is to be done though.  Should I just remove the \usage line?

Reply:
I like the idea.

You can use doxygen's `\code{.<language_extension>}` / `\endcode` directives to add code sections/examples to the documentation. For example, using Python styling:
```doxygen
\code{.py}
writeLine("stdout", "This shows up on stdout");
writeLine("stderr", "This shows up on stderr");
writeLine("serverLog", "This shows up in the server log");
writeLine("/tempZone/home/alice/foo", "This is written to the data object /tempZone/home/alice/foo, I guess");
\endcode
```

Reply:
And yes, remove the `\usage` line and open an issue to replace all of them with examples.

Reply:
I removed the \usage line.

I would have added the code but I wasn't sure of the following:

1. Would that code follow the \usage tag?
2. If I used `\code{.re}` would that work?

Reply:
> 1. Would that code follow the \usage tag?
> 2. If I used `\code{.re}` would that work?

See https://github.com/irods/irods/issues/7630#issuecomment-2037559883.

Reply:
Let's go ahead and add the code examples. It's just 6 lines.

Reply:
> Let's go ahead and add the code examples. It's just 6 lines.

Done